### PR TITLE
Migrate to bullseye debian base image

### PR DESCRIPTION
**Pull request by saikat-royc**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Move to the bullseye debian base image which contains the CVE fixes

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Tested volume mounts/unmounts against both arm and x86 GKE nodes

**Does this PR introduce a user-facing change?**:

```release-note
Migrate to bullseye debian base image
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -45,10 +45,13 @@ ENV DEBIAN_FRONTEND noninteractive
 # between older and newer distros (in this case it was /etc/rpc
 # existing only in the old launcher.gcr.io image) and using dpgk -S
 # <file> to determine which package supplies it, can be helpful.
-RUN apt-get update && apt-get dist-upgrade -y && apt-get install -y --no-install-recommends \
+RUN apt-get update && apt-get dist-upgrade -y && apt-mark unhold libcap2 && apt-get install --reinstall -y --no-install-recommends \
```
Comment (wespanther) [inline Dockerfile:48]: For future reference, we might add a comment about why we need to unhold libcap2 and add the `--reinstall` flag.

**Review by wespanther — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (saikat-royc) [reply Dockerfile:54]: Done

**Pull request merged**